plotting/plot_handler.py

```diff
@@ -0,0 +1,107 @@
+            self.rb_number) + "/" + str(self.run_number) + "/autoreduced/"
+        # parameter to store the file names of any existing plot files found
+        self._existing_plot_files = []
+
+    def _regexp_for_file_name(self, plot_type):
+        """
+        Regular expression used for looking for plot files. It assumes that the file names follow the convention
+        <instrument_abbreviation><run_number>*<.png or other extension>
+        """
+        try:
+            _regexp = self._instrument_dict[self.instrument_name] + str(
+                self.run_number) + "*." + plot_type
+            return _regexp
+        except KeyError:  # if the instrument name does not appear in the dictionary of known instruments
+            LOGGER.info("The instrument name is not recognised")
+
+    def _check_for_plot_files(self):
+        """
+        Searches the CEPH directory for existing plot files using the directory specified by instrument_name, rb_number,
+        run_number, and a regular expression for the file name based on the instrument_name, run_number and plot_type.
+        """
+        # start sftpclient
+        client = SFTPClient()
+        # initialise list to store names of existing files matching the search
+        _found_files = []
+        for plot_type_i in self.plot_type:
+            # directory where existing plot file(s) would be located
+            ceph_dir = self._rbfolder
+            # regular expression for plot file name(s)
+            file_regex = self._regexp_for_file_name(plot_type=plot_type_i)
+            # use sftpclient to check if files matching the regular expression exist and add any matches to the list
+            _found_files.extend(client.get_filenames(server_dir_path=ceph_dir, regex=file_regex))
+        return _found_files
+
+    def get_plot_file(self):
+        """
+        Searches for and retrieves a plot file from CEPH. Might find multiple files (e.g. if more than one plot_type is
+        specified), but will only copy over one. If no existing plot file is found it does nothing at the moment.
+        """
+        self._existing_plot_files = self._check_for_plot_files()
+        if len(self._existing_plot_files) == 0:  # no existing file was found
+            return []  # return an empty list for now
+        else:  # if one or more existing files were found, use the first one (order of items in plot_type affects this)
+            _ceph_path = self._rbfolder + self._existing_plot_files[0]
+            # in case the local path to which the file gets copied to needs to be specified
+            _local_path = None
+            # create sftpclient object and try to retrieve the file
+            client = SFTPClient()
+            try:
+                client.retrieve(server_file_path=_ceph_path, local_file_path=_local_path, override=True)
+            except RuntimeError:
+                logging.error("file does not exist")
```

Comment:
Because you've saved your logger as `LOGGER` (line 17), I believe **all** log statements need to begin with
`LOGGER.` rather than `logging.`

Reply:
both actually work but we should use `LOGGER` to be consistent

Reply:
good catch. changed it.